cluster-autoscaler/cloudprovider/gce/autoscaling_gce_client.go

```diff
@@ -453,15 +453,18 @@ func isReservationNotReady(errorCode, errorMessage string) bool {
 }
 
 func isInvalidReservationError(errorCode, errorMessage string) bool {
-	reservationErrors := []string{
-		"Incompatible AggregateReservation VMFamily",
-		"Could not find the given reservation with the following name",
-		"must use ReservationAffinity of",
-		"The reservation must exist in the same project as the instance",
-		"only compatible with Aggregate Reservations",
+	regexReservationErrors := []*regexp.Regexp{
+		regexp.MustCompile("Incompatible AggregateReservation VMFamily"),
+		regexp.MustCompile("Could not find the given reservation with the following name"),
+		regexp.MustCompile("must use ReservationAffinity of"),
+		regexp.MustCompile("The reservation must exist in the same project as the instance"),
+		regexp.MustCompile("only compatible with Aggregate Reservations"),
+		regexp.MustCompile("instance configuration is optimized for ([a-z0-9*-]+) workloads"),
```

Comment:
Please use the other part of the error message: "Please target a reservation with workload_type" (makes it clear that the error is about reservations and what's it about).

Reply:
Done